App.tsx

```diff
@@ -77,7 +83,14 @@ class App extends Component<Props, State> {
   async componentDidMount() {
     TrackPlayer.registerPlaybackService(() => require('./src/services/playerAudioEvents'))
     await PlayerAudioSetupService()
-    
+    // Android Auto
+    if (Platform.OS === 'android') {
+      // initialize Android Auto Tabs with no content. Content will be updated as they are loaded to the global state.
+      registerAndroidAutoModule()
+      PVEventEmitter.on(PV.Events.QUEUE_HAS_UPDATED, handleAndroidAutoQueueUpdate)
+      PVEventEmitter.on(PV.Events.APP_FINISHED_INITALIZING_FOR_CARPLAY, onAppInitialized)
+    }
```

Comment:
Any way we could utilize the onConnect method for this and not just register in componentDidMount? I don't think we need to startup android auto processes if the app isn't connected to a car.

I do see your conditional for calling registerCarModule but I think that's where this code should go in. Just to keep some consistency and not do too many things different per platform.

Reply:
Yes... to register an onConnect function android requires to add a broadcast receiver, though it might have problems when app is in background/cold start.
https://stackoverflow.com/questions/48402194/how-to-detect-appor-mobile-is-connected-to-android-auto
Technically what we are building here, MediaBrowserService is meant to make the entire media content browsable by any other app, making it available at all times is the most common practice. I guess this is not really the case for iOS where a template is being pushed once connected? 
All the relevant event listeners and content updates should be android only and even android auto only, as they are called inside the Event.RemoteBrowse listener.